lib/allora/README.md

```diff
@@ -0,0 +1,134 @@
+
+Multiple processes run on the Allora Worker Node (EC2 instance):
+
+  - Docker container with the worker node logic that handles communnication bet
+  - Docker container running the model server that reveal inferences to consumers
+
+Allora Public Head Nodes publish the Allora chain requests (requests for inferences from consumers) to Allora worker nodes. When a worker node is initialized, it starts with an environment variable called BOOT_NODES, which helps handle the connection and communications between worker nodes and the head nodes.
+
+The worker node (docker container) will call the function that invokes custom logic that handles. The request-response is a bidirectional flow from the Allora chain (inference requests from consumers) to the public head nodes to the worker node and finally to the model server that reveals inferences. 
+
+
+## Worker Node System Requirements
+
+- Operating System: Any modern Linux operating system
+- CPU: Minimum of 1/2 core
+- Memory: 2 to 4 GB
+- Storage: SSD or NVMe with at least 5GB of space
+
+## Setup Instructions
+
+### Setup Cloud9
+
+We will use AWS Cloud9 to execute the subsequent commands. Follow the instructions in [Cloud9 Setup](../../docs/setup-cloud9.md).
+
+### Clone this repository and install dependencies
+
+```bash
+   git clone https://github.com/aws-samples/aws-blockchain-node-runners.git
+   cd aws-blockchain-node-runners
+   npm install
+```
+
+### Deploy single worker node
+
+1. Make sure you are in the root directory of the cloned repository
+
+2. Configure your setup
+
+    Create your own copy of `.env` file and edit it to update with your AWS Account ID and Region:
+    ```bash
+   # Make sure you are in aws-blockchain-node-runners/lib/allora
+   cd lib/allora
+   npm install
+   pwd
+   cp ./sample-configs/.env-sample-full .env
+   nano .env
+    ```
+   > NOTE:
+   > Example configuration parameters are set in the local `.env-sample` file. You can find more examples inside `sample-configs` directory.
+
+   > IMPORTANT:
+   > All AWS CDK v2 deployments use dedicated AWS resources to hold data during deployment. Therefore, your AWS account and Region must be [bootstrapped](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html) to create these resources before you can deploy. If you haven't already bootstrapped, issue the following command:
+   > ```bash
+   > cdk bootstrap aws://ACCOUNT-NUMBER/REGION
+   > ```
+
+3. Deploy Common Stack
+
+   ```bash
+   pwd
+   # Make sure you are in aws-blockchain-node-runners/lib/allora
+   npx cdk deploy allora-edge-common --json --outputs-file allora-edge-common-deploy.json
+   ```
+
+5. Deploy Allora Worker Node
+
+   ```bash
+   pwd
+   # Make sure you are in aws-blockchain-node-runners/lib/allora
+   npx cdk deploy allora-single-node --json --outputs-file single-node-deploy.json
```

Comment:
When I run this, I get this error (tried it twice). I'm not clear which resource it is failing to create. Are you able to run this step successfully?

```2024-07-17 13:37:02 UTC-0400
allora-single-node
ROLLBACK_IN_PROGRESS
-
The following resource(s) failed to create: [AlloraWorkerxSingleNodesinglenode2A6E247F, AlloraWorkerxEIPAssociation]. Rollback requested by user.
2024-07-17 13:37:01 UTC-0400
AlloraWorkerxEIPAssociation
CREATE_FAILED
-
Resource creation cancelled
2024-07-17 13:36:58 UTC-0400
AlloraWorkerxSingleNodesinglenode2A6E247F
CREATE_FAILED
-
Failed to receive 1 resource signal(s) within the specified duration```

Reply:
There was a problem running cfn-signal on this particular AMI (ubuntu). I have since migrated to Amazon Linux 2 (and adjusted user-data), and it seems to be working fine.